### PR TITLE
Add metrics for restic back up operation

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Signed-off-by: Ashish Amarnath <ashisham@vmware.com>

Changes in this PR:
1. Start a metrics server in the restic daemonset pods.
1. Create and register the following metrics for:
    - total number of pod volume backups enqueued
    - total number of pod volume backups processed
    - Latency of restic backup operation both as a histogram and a gauge.
1. Publish metrics in the pod_volume_backup_controller.

### CODE REVIEW

**Comment by nrb** [comment]
@ashish-amarnath Is this still WIP? IMO, if we keep the current number of metrics, we could get it in for 1.5.

**Comment by ashish-amarnath** [comment]
I think we should get this PR merged and add more metrics. Removing WIP.

**Comment by stephbman** [comment]
Let's talk about this one in the backlog grooming this afternoon

**Comment by ashish-amarnath** [comment]
@stephbman Is the discussion a pre-requisite to having this PR merged?

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -242,6 +254,56 @@ func NewServerMetrics() *ServerMetrics {
...
+					Buckets: []float64{
+						toSeconds(1 * time.Minute),
+						toSeconds(5 * time.Minute),
+						toSeconds(10 * time.Minute),
+						toSeconds(15 * time.Minute),
+						toSeconds(30 * time.Minute),
+						toSeconds(1 * time.Hour),
+						toSeconds(2 * time.Hour),
+						toSeconds(3 * time.Hour),
+						toSeconds(4 * time.Hour),
```
Comment (nrb) [inline pkg/metrics/metrics.go:298]: What happens if an operation takes longer than 4 hours? Does it go into the 4 hour bucket, or is a new one made?

Comment (ashish-amarnath) [reply pkg/metrics/metrics.go:298]: In this case 10 buckets are created for the following ranges:
1. 0-60s
1. 0-300s
1. 0-600s
1. 0-900s
1. 0-1800s
1. 0-3600s
1. 0-7200s
1. 0-10800s
1. 0-14400s
1. 0-+Inf

All the operations taking longer than 4h will fall into the last bucket [0-+Inf]

Comment (stephbman) [reply pkg/metrics/metrics.go:298]: @ashish-amarnath please feel free to merge the PR :)

**Pull request merged**